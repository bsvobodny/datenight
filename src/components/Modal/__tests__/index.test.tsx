import { render, screen, fireEvent } from '@testing-library/react'
import { describe, it, expect, vi } from 'vitest'
import '@testing-library/jest-dom'
import { Modal } from '../index'

describe('Modal Component', () => {
  it('renders children when open', () => {
    render(
      <Modal isOpen={true} onClose={() => {}} title="Modal Title">
        <div>Modal Content</div>
      </Modal>
    )
    expect(screen.getByText('Modal Content')).toBeInTheDocument()
  })

  it('does not render children when closed', () => {
    render(
      <Modal isOpen={false} onClose={() => {}} title="Modal Title">
        <div>Hidden Content</div>
      </Modal>
    )
    expect(screen.queryByText('Hidden Content')).not.toBeInTheDocument()
  })

  it('calls onClose when backdrop is clicked', () => {
    const onClose = vi.fn()
    render(
      <Modal isOpen={true} onClose={onClose} title="Modal Title">
        <div>Modal Content</div>
      </Modal>
    )
    fireEvent.click(screen.getByTestId('modal-overlay'))
    expect(onClose).toHaveBeenCalled()
  })

  it('calls onClose when Escape key is pressed', () => {
    const onClose = vi.fn()
    render(
      <Modal isOpen={true} onClose={onClose} title="Modal Title">
        <div>Modal Content</div>
      </Modal>
    )
    fireEvent.keyDown(document, { key: 'Escape', code: 'Escape' })
    expect(onClose).toHaveBeenCalled()
  })

  it('matches snapshot when open', () => {
    const { asFragment } = render(
      <Modal isOpen={true} onClose={() => {}} title={'Modal Title'}>
        <div>Snapshot Content</div>
      </Modal>
    )
    expect(asFragment()).toMatchSnapshot()
  })
})
