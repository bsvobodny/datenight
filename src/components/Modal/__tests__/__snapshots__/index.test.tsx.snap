// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Modal Component > matches snapshot when open 1`] = `
<DocumentFragment>
  <div
    class="modal-overlay modal-open"
    data-testid="modal-overlay"
  >
    <div
      class="modal-content modal-medium"
    >
      <div
        class="modal-header"
      >
        <h2
          class="modal-title"
        >
          Modal Title
        </h2>
        <button
          aria-label="Fermer"
          class="modal-close-btn"
        >
          X
        </button>
      </div>
      <div
        class="modal-body"
      >
        <div>
          Snapshot Content
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
